### PR TITLE
Gradle build

### PR DESCRIPTION
**Pull request by jburel**
Build application using gradle
By default 5.2.1 version is used. This is the version currently used in the various gradle projects.

### CODE REVIEW

File: workflow-templates/gradle.properties.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "name": "Build using Gradle",
+    "description": "Build using Gradle, by default version 5.2.1 is used.",
+    "iconName": "gradle"
```
Comment (sbesson) [inline workflow-templates/gradle.properties.json:4]: As per https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/sharing-workflows-with-your-organization#creating-a-workflow-template:
-  proposing to add `Gradle` and/or `Java` as the `Categories` - see https://github.com/github/linguist/blob/master/lib/linguist/languages.yml
- can we make assumptions about the `filePatterns` e.g. `build.gradle` or `*.gradle` in the top-level folder?
filePatterns

**Review by sbesson — COMMENTED** [review]
Overall, looks very nice. A few stylistic comments about the JSON file.

Do you have a shortlist of the candidate repositories for rolling this template over? Have you pretested it on a few of them already?

**Comment by jburel** [comment]
I have tested it on https://github.com/jburel/omero-model/actions/runs/366626900
This will be applied to most of the repositories created when we decoupled openmicroscopy.
A couple will require some adjustments e.g. omero-blitz

File: workflow-templates/gradle.properties.json (hunk 1 above)
Comment (jburel) [reply workflow-templates/gradle.properties.json:4]: build.gradle is in the top-level folder of our repositories.

**Comment by sbesson** [comment]
Other Gradle repos I am aware of are:

- bio-formats-examples
- minimal-omero-client
- omero-gradle-plugins

I can have a go at testing the two first ones next week. Unsure whether the latter will need a separate logic

**Comment by jburel** [comment]
I was considering omero-gradle-plugins as part of the decoupled effort

**Comment by jburel** [comment]
minimal-omero-client: a Dockerfile is used to run gradle build

**Comment by jburel** [comment]
For bio-formats-example: we have a combination of Java version so I will add Java to the template.

File: workflow-templates/gradle.properties.json (hunk 1 above)
Comment (jburel) [reply workflow-templates/gradle.properties.json]: added ``Categories`` and ``filePatterns``

**Review by sbesson — APPROVED** [review]
Thanks for the changes. Happy for this to be added to the templates as it is and start rolling it over to the various Gradle repositories unless @joshmoore thinks of something else.

**Comment by joshmoore** [comment]
Looks good. There may still be some repositories with wrapper logic and/or hard-coded versions in various repos that don't like being told what version to use, but we can deal with those by stripping out those assumptions. :+1:

**Comment by sbesson** [comment]
Screenshot 2020-11-18 at 13 37 11 👍

**Pull request merged**